src/main/java/sirius/db/jdbc/SQLQuery.java

```diff
@@ -127,39 +127,51 @@ public void iterate(Function<Row, Boolean> handler, @Nullable Limit limit) throw
                 if (stmt == null) {
                     return;
                 }
-                if (limit == null) {
-                    limit = Limit.UNLIMITED;
-                }
-                if (limit.getTotalItems() > 0) {
-                    stmt.setMaxRows(limit.getTotalItems());
-                }
-                if (limit.getTotalItems() > 1000 || limit.getTotalItems() <= 0) {
-                    if (ds.hasCapability(Capability.STREAMING)) {
-                        stmt.setFetchSize(Integer.MIN_VALUE);
-                    } else {
-                        stmt.setFetchSize(1000);
-                    }
-                }
+                Limit effectiveLimit = (limit == null) ? Limit.UNLIMITED : limit;
+                applyMaxRows(stmt, effectiveLimit);
+                applyFetchSize(stmt, effectiveLimit);
+
                 try (ResultSet rs = stmt.executeQuery()) {
                     TaskContext tc = TaskContext.get();
-                    while (rs.next() && tc.isActive()) {
-                        Row row = loadIntoRow(rs);
-                        if (limit.nextRow()) {
-                            if (!handler.apply(row)) {
-                                break;
-                            }
-                        }
-                        if (!limit.shouldContinue()) {
-                            break;
-                        }
-                    }
+                    processResultSet(handler, effectiveLimit, rs, tc);
                 }
             }
         } finally {
             w.submitMicroTiming("SQL-QUERY", sql);
         }
     }
 
+    protected void applyMaxRows(PreparedStatement stmt, Limit effectiveLimit) throws SQLException {
+        if (effectiveLimit.getTotalItems() > 0) {
+            stmt.setMaxRows(effectiveLimit.getTotalItems());
+        }
+    }
+
+    protected void applyFetchSize(PreparedStatement stmt, Limit effectiveLimit) throws SQLException {
+        if (effectiveLimit.getTotalItems() > 1000 || effectiveLimit.getTotalItems() <= 0) {
+            if (ds.hasCapability(Capability.STREAMING)) {
+                stmt.setFetchSize(Integer.MIN_VALUE);
+            } else {
+                stmt.setFetchSize(1000);
+            }
+        }
+    }
+
+    protected void processResultSet(Function<Row, Boolean> handler, Limit effectiveLimit, ResultSet rs, TaskContext tc)
+            throws SQLException {
+        while (rs.next() && tc.isActive()) {
+            Row row = loadIntoRow(rs);
+            if (effectiveLimit.nextRow()) {
+                if (!handler.apply(row)) {
```

Comment:
das ist ja der _eigentliche_ aufruf, deswegen ists wenigstens in ner eigenen zeile